packages/@expo/cli/src/start/server/metro/inspector-proxy/proxy.ts

```diff
@@ -0,0 +1,155 @@
+import type { Server as HttpServer, IncomingMessage, ServerResponse } from 'http';
+import type { Server as HttpsServer } from 'https';
+import type { InspectorProxy as MetroProxy, Device as MetroDevice } from 'metro-inspector-proxy';
+import { parse } from 'url';
+import WS, { Server as WSServer } from 'ws';
+
+const WS_DEVICE_URL = '/inspector/device';
+const WS_DEBUGGER_URL = '/inspector/debug';
+const WS_GENERIC_ERROR_STATUS = 1011;
+
+const debug = require('debug')('expo:metro:inspector-proxy:proxy') as typeof console.log;
+
+// This is a workaround for `ConstructorType` not working on dynamically generated classes
+type Instantiatable<Instance> = new (...args: any) => Instance;
+
+export class ExpoInspectorProxy<D extends MetroDevice = MetroDevice> {
+  constructor(
+    public readonly metroProxy: MetroProxy,
+    private DeviceClass: Instantiatable<D>,
+    public readonly devices: Map<number, D> = new Map()
+  ) {
+    this.metroProxy = metroProxy;
```

Comment:
```suggestion
```
I don't think this is required, the `public` part of the initializer should bind it to the class scope.